File: solutions/javascript/bob/2/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const hey = (message) => {
  let trimmed = message.trim()

  const isEmpty = trimmed.length === 0
  const isQuestion = trimmed.endsWith('?')
  const isShout = /[a-z]/i.test(trimmed) && trimmed === trimmed.toUpperCase()
  
  switch (true) {
    case isQuestion && isShout:
      return "Calm down, I know what I'm doing!";
    case isShout:
      return 'Whoa, chill out!';
    case isQuestion:
      return 'Sure.';
    case isEmpty:
      return 'Fine. Be that way!'
    default:
      return 'Whatever.';
  }
};

File: solutions/fortran/bob/1/bob.f90
module bob
  implicit none
contains
  function hey(statement)
    character(100) :: hey
    character(len=*), intent(in) :: statement
    logical :: is_question
    logical :: is_yelling
    is_question = (index(trim(statement), '?') == len(trim(statement)))
    is_yelling = (scan(statement, 'abcdefghijklmnopqrstuvwxyz') == 0) .and. (scan(statement, "ABCDEFGHIJKLMNOPQRSTUVWXYZ") /= 0) 
    if (is_question .and. is_yelling) then
        hey = "Calm down, I know what I'm doing!"
    endif
    if (is_question .and. .not. is_yelling) then
        hey = "Sure."
    endif
    if (.not. is_question .and. is_yelling) then
        hey = "Whoa, chill out!"
    endif
    if (.not. is_question .and. .not. is_yelling) then
        hey = "Whatever."
    endif
    if (len(trim(statement)) == 0) then 
      hey = "Fine. Be that way!" 
    endif
  end function hey
end module bob